src/Build.UnitTests/BackEnd/TargetBuilder_Tests.cs

```diff
@@ -1254,6 +1254,31 @@ public void TestCircularDependencyInCallTarget()
             Assert.False(success);
         }
 
+        /// <summary>
+        /// Tests a circular dependency target.
+        /// </summary>
+        [Fact]
+        public void TestCircularDependencyTarget()
+        {
+            string projectContents = @"
+<Project xmlns=""http://schemas.microsoft.com/developer/msbuild/2003"">
+    <Target Name=""TargetA"" AfterTargets=""Build"" DependsOnTargets=""TargetB"">
+        <Message Text=""TargetA""></Message>
+    </Target>
+    <Target Name=""TargetB"" DependsOnTargets=""TargetC"">
+        <Message Text=""TargetB""></Message>
+    </Target>
+    <Target Name=""TargetC"" DependsOnTargets=""TargetA"">
+        <Message Text=""TargetC""></Message>
+    </Target>
+</Project>
+      ";
+            StringReader reader = new StringReader(projectContents);
+            Project project = new Project(new XmlTextReader(reader), null, null);
+            bool success = project.Build(_mockLogger);
+            Assert.False(success);
```

Comment:
It would be good to verify that the reason it's failing is as expected, and the parents are properly printed.

Reply:
Make sense. I figure out a way to check the error message.